src/components/DataCollector/activity/click.js

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const isValidObject = clickObj => {
+  if (clickObj) {
+    if (clickObj.tagName || clickObj.parentElement || clickObj.parentNode) {
+      return true;
+    }
+  }
+  return false;
+};
+
+const isSupportedHrefObject = obj => {
+  if (
+    obj.href &&
+    (obj.tagName === "A" || obj.tagName === "AREA") &&
+    (!obj.onclick ||
+      !obj.protocol ||
+      obj.protocol.toLowerCase().indexOf("javascript") < 0)
+  ) {
+    return true;
+  }
+  return false;
+};
+
+const getObjectHref = obj => {
+  const loc = window.location;
+  let href = obj.href ? obj.href : "";
+  let { protocol, host } = obj;
+  let i = href.indexOf(":");
+  const j = href.indexOf("?");
+  const k = href.indexOf("/");
+  if (href && (i < 0 || (j >= 0 && i > j) || (k >= 0 && i > k))) {
```

Comment:
We need a way to make this `if` clearer somehow: either we place the condition in a variable with a meaningful name or even break it down into multiple variables.